Firestore/core/src/firebase/firestore/local/serializer.cc

```diff
@@ -140,6 +140,24 @@ std::unique_ptr<model::MaybeDocument> Serializer::DecodeMaybeDocument(
   }
 }
 
+void Serializer::EncodeDocument(Writer* writer, const model::Document& doc) const {
+  // Encode Document.name
+  writer->WriteTag({PB_WT_STRING, google_firestore_v1beta1_Document_name_tag});
```

Comment:
Can this implementation take advantage of the fact that the format is the same as `remote::Serializer::EncodeDocument` with an additional field appended? Something like:
```cpp
rpc_serializer_.EncodeDocument(writer, doc);
writer->WriteTag({PB_WT_STRING, google_firestore_v1beta1_Document_update_time_tag});
writer->WriteNestedMessage([&](Writer* writer) {
  rpc_serializer_.EncodeVersion(writer, doc.version());
});
```

Reply:
Splitting parsing of a message like that makes me a little nervous, but I think it should work perfectly fine. (Decoding wouldn't though.)

OTOH, it wouldn't be saving much, and these methods are logically distinct. (The local serializer theoretically shouldn't use *anything* from the remote serializer as the on-disk protos and the on-wire protos are considered different. Even though they largely aren't.) So the long term, probably never reached end point, is that none of these methods call anything in the remote serializer.

Left as is for now.

Reply:
I'm fine either way, thanks for the explanation.